mirrord/sip/src/lib.rs

```diff
@@ -480,7 +486,28 @@ mod main {
             return Ok(NoSip);
         }
 
-        if let Some(shebang) = read_shebang_from_file(&complete_path)? {
+        // determine if file is object file
+        let data = std::fs::read(&complete_path)?;
```

Comment:
We load the target binary in it's entirety twice into memory here because the data is only passed as ref and not consumed by it, I don't think this is an issue in most cases unless the target binary is huge for some reason (multiple hundreds of MB or even several GB). I would take a look at maybe changing `is_binary_sip ` & `is_code_signed` to use the `MachFile` directly instead of paths (or with both if necessary) instead of creating them inside of them